libvips/foreign/dcrawload.c

```diff
@@ -310,6 +517,62 @@ vips_foreign_load_dcraw_file_init(VipsForeignLoadDcRawFile *file)
 {
 }
 
+typedef struct _VipsForeignLoadRawBuffer {
+	VipsForeignLoadDcRaw parent_object;
+
+	/* Load from a buffer.
+	 */
+	VipsBlob *blob;
+
+} VipsForeignLoadDcRawBuffer;
+
+typedef VipsForeignLoadDcRawClass VipsForeignLoadDcRawBufferClass;
+
+G_DEFINE_TYPE(VipsForeignLoadDcRawBuffer, vips_foreign_load_dcraw_buffer,
+	vips_foreign_load_dcraw_get_type());
+
+static int
+vips_foreign_load_dcraw_buffer_build(VipsObject *object)
+{
+	VipsForeignLoadDcRaw *raw = (VipsForeignLoadDcRaw *) object;
+	VipsForeignLoadDcRawBuffer *buffer = (VipsForeignLoadDcRawBuffer *) object;
+
+	if (buffer->blob &&
+		!(raw->source = vips_source_new_from_memory(
+			  VIPS_AREA(buffer->blob)->data,
+			  VIPS_AREA(buffer->blob)->length)))
+		return -1;
+
+	return VIPS_OBJECT_CLASS(vips_foreign_load_dcraw_buffer_parent_class)
+		->build(object);
+}
+
+static void
+vips_foreign_load_dcraw_buffer_class_init(
+	VipsForeignLoadDcRawBufferClass *class)
+{
+	GObjectClass *gobject_class = G_OBJECT_CLASS(class);
+	VipsObjectClass *object_class = (VipsObjectClass *) class;
+
+	gobject_class->set_property = vips_object_set_property;
+	gobject_class->get_property = vips_object_get_property;
+
+	object_class->nickname = "rawload_buffer";
+	object_class->build = vips_foreign_load_dcraw_buffer_build;
+
```

Comment:
Ditto, perhaps we should a `->is_a_buffer` here that always returns `false`. For example:
```c
/* Compile with:
 *   gcc -g -Wall test.c `pkg-config vips --cflags --libs`
 */

#include <vips/vips.h>

int
main(int argc, char *argv[])
{
	if (VIPS_INIT(argv[0]))
		vips_error_exit(NULL);

	VipsSource *source;
	if (!(source = vips_source_new_from_file("/dev/null")))
		return 0;

	vips_foreign_find_load_source(source);
	vips_foreign_find_load_buffer(NULL, 0);

	return 0;
}
```

```console
$ ./a.out 

(process:33687): VIPS-WARNING **: 12:39:48.840: loader dcrawload_source has no is_a_source method

(process:33687): VIPS-WARNING **: 12:39:48.841: loader dcrawload_buffer has no is_a_buffer method
```